File: usecase/templates/app-java/src/main/java/usecase/_UseCaseImpl.java
package <%= appPackage %>.domain.usecases.<%= useCasePackageName %>;

import rx.Scheduler;
import <%= appPackage %>.domain.interactors.base.BaseUseCase;
import rx.Observable;

public class <%= useCaseName %>UseCaseImpl extends BaseUseCase implements <%= useCaseName %>UseCase {

    public <%= useCaseName %>UseCaseImpl(Scheduler executor) {
        super(executor);
    }

    @Override
    public Observable<Object> invoke() {
        return Observable.just(null);
    }

}

File: usecase/templates/app-kotlin/src/main/java/usecase/_UseCaseImpl.kt
package <

%= appPackage %>.domain.usecases.<%= useCasePackageName %>;

import <%= appPackage %>.domain.interactors.base.BaseUseCase;

import rx.Scheduler;
import rx.Observable;

class <%= useCaseName %>UseCaseImpl(executor: Scheduler) : BaseUseCase(executor), <%= useCaseName %>UseCase {

    override fun invoke(): Observable<Any> {
        return Observable.just(null);
    }

}
